actor/builtin/miner/miner.go

```diff
@@ -950,8 +950,7 @@ func (ma *Actor) SubmitPoSt(ctx exec.VMContext, poStProofs []types.PoStProof, do
 		// Update miner power to the amount of data actually proved
 		// during the last proving period.
 		oldPower := state.Power
-		// TODO subtract total faulted size from ProvingSet size #2889
-		newPower := types.NewBytesAmount(uint64(state.ProvingSet.Size())).Mul(state.SectorSize)
+		newPower := types.NewBytesAmount(uint64(state.ProvingSet.Size() - faults.SectorIds.Size())).Mul(state.SectorSize)
 		state.Power = newPower
 		delta := newPower.Sub(oldPower)
 		_, ret, err := ctx.Send(address.StorageMarketAddress, "updateStorage", types.ZeroAttoFIL, []interface{}{delta})
```

Comment:
I know this wasn't your doing, but we really should skip this call if the delta is zero. Would you mind adding that?

Reply:
Feel free to add a follow on "good first" issue alternatively